### ISSUE TITLE
nextOpen function: Xdebug has detected a possible infinite loop, and aborted your script with a stack depth of '256' frames

### ISSUE THREAD

**Issue by BenMaldoner** [spatie/opening-hours#255]
We use the latest spatie/opening-hours version (4.0.1) and the following code will create a possible infinity loop and is aborted with the error message 

> Xdebug has detected a possible infinite loop, and aborted your script with a stack depth of '256' frames

:

```
   public function test_infinity_loop()
   {
      $ohConfig = ["monday" => ["00:01-00:00"], "tuesday" => ["00:01-00:00"], "wednesday" => ["00:01-00:00"], "thursday" => ["00:01-00:00"], "friday" => ["00:01-00:00"], "saturday" => ["00:01-00:00"], "sunday" => ["00:01-00:00"]];
      $openingHoursObject = OpeningHours::create($ohConfig);
      $nextOpen = $openingHoursObject->nextOpen(new Carbon("2024-10-22 00:00:00"));
      $this->assertNotNull($nextOpen);
   }
```

And this error also persists if you increase the xdebug.max_nesting_level (e.g. 1024).

I know that the input data should probably be "00:00-23:59" instead of "00:01-00:00" but we get the opening hours from an external data provider and sometimes people create opening hours like this. So it would be great if it was handled by the lib.

**Comment by kylekatarnls** [comment]
Hello,

For the record "00:00-23:59" means that the business is closed between 23:59:00 and 23:59:59.999999, and open the rest of the day, "00:01-00:00" means that the business is closed between 00:00:00 and 00:00:59.999999 and closed the rest of the day.

To mean open all day long the correct syntax is: "00:00-24:00".

Infinite loops should be protected by `MaximumLimitExceeded` and `SearchLimitReached` which are thrown if iterating for too long without finding open/close change and `MaximumLimitExceeded` should be triggered when calling nextOpen for a business always open.

It sounds we have indeed a bug to handle 00:01-00:00 which is not happening for any other config, I will investigate this.

But for the record when start > end (i.e. 00:01 > 00:00) it's considered as an overflow to the next day which has a particular handling (see `'overflow' => true` in the readme).

So it's probably good to have some validation of the data before you ingest it. Either providing error then guidance for the third-party of what to put, or it can detect and fix if it detects the intent is "open all day long".

**Comment by kylekatarnls** [comment]
It was actually an easy fix, it's now solved in 4.0.2

**Comment by BenMaldoner** [comment]
Thanks a lot for the quick fix! Works!
